SNOPT tests failing on stable images

<!-- Please search on GitHub first to see if anyone else has already opened the same issue -->
## Description
With changes from PR https://github.com/mdolab/pyoptsparse/pull/364, SNOPT tests are failing on stable images during nightly tests with `fStar` reported as 0 (see for example [here](https://github.com/mdolab/docker/actions/runs/7096395939/job/19314852484#step:13:13471)).

### Steps to reproduce issue
Any stable image with changes from https://github.com/mdolab/pyoptsparse/pull/364 will generate similar errors as below

```
Traceback (most recent call last):
  File "/home/mdolabuser/repos/pyoptsparse/tests/test_tp109.py", line 205, in test_autorefine
    self.assert_solution_allclose(sol2, 1e-2)
  File "/home/mdolabuser/repos/pyoptsparse/tests/testing_utils.py", line 123, in assert_solution_allclose
    assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
  File "/home/mdolabuser/.pyenv/versions/3.9.16/lib/python3.9/site-packages/numpy/testing/_private/utils.py", line 1530, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/home/mdolabuser/.pyenv/versions/3.9.16/lib/python3.9/site-packages/numpy/testing/_private/utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=0.01, atol=0.01

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 5362.06927538
Max relative difference: 1.
 x: array(0.)
 y: array(5362.069275)
```

### Current behavior
SNOPT tests on stable images are failing

### Expected behavior
All tests should pass

### Code versions
Any stable image with changes from https://github.com/mdolab/pyoptsparse/pull/364

--- a/pyoptsparse/pySNOPT/pySNOPT.py
+++ b/pyoptsparse/pySNOPT/pySNOPT.py
@@ -521,9 +521,11 @@ class SNOPT(Optimizer):
             sol_inform["text"] = self.informs[inform]
 
             # Create the optimization solution
-            if parse_version(self.version) > parse_version("7.7.0") and parse_version(self.version) < parse_version("7.7.7"):
+            if parse_version(self.version) > parse_version("7.7.0") and parse_version(self.version) < parse_version(
+                "7.7.7"
+            ):
                 # SNOPT obj value is buggy and returned as 0, its thus overwritten with the solution objective value
-                obj = np.array([obj.value*obj.scale for obj in self.optProb.objectives.values()])
+                obj = np.array([obj.value * obj.scale for obj in self.optProb.objectives.values()])
 
             sol = self._createSolution(optTime, sol_inform, obj, xs[:nvar], multipliers=pi)
             restartDict = {

--- a/pyoptsparse/pySNOPT/pySNOPT.py
+++ b/pyoptsparse/pySNOPT/pySNOPT.py
@@ -15,6 +15,7 @@ import warnings
 from baseclasses.utils import CaseInsensitiveSet
 import numpy as np
 from numpy import ndarray
+from pkg_resources import parse_version
 
 # Local modules
 from ..pyOpt_error import Error
@@ -520,7 +521,14 @@ class SNOPT(Optimizer):
             sol_inform["text"] = self.informs[inform]
 
             # Create the optimization solution
-            sol = self._createSolution(optTime, sol_inform, obj, xs[:nvar], multipliers=pi)
+            if parse_version(self.version) > parse_version("7.7.0") and parse_version(self.version) < parse_version(
+                "7.7.7"
+            ):
+                # buggy SNOPT obj value
+                sol_objectives = np.array([obj.value for obj in self.optProb.objectives.values()])
+                sol = self._createSolution(optTime, sol_inform, sol_objectives, xs[:nvar], multipliers=pi)
+            else:
+                sol = self._createSolution(optTime, sol_inform, obj, xs[:nvar], multipliers=pi)
             restartDict = {
                 "cw": cw,
                 "iw": iw,

--- a/pyoptsparse/pySNOPT/pySNOPT.py
+++ b/pyoptsparse/pySNOPT/pySNOPT.py
@@ -521,14 +521,11 @@ class SNOPT(Optimizer):
             sol_inform["text"] = self.informs[inform]
 
             # Create the optimization solution
-            if parse_version(self.version) > parse_version("7.7.0") and parse_version(self.version) < parse_version(
-                "7.7.7"
-            ):
-                # buggy SNOPT obj value
-                sol_objectives = np.array([obj.value for obj in self.optProb.objectives.values()])
-                sol = self._createSolution(optTime, sol_inform, sol_objectives, xs[:nvar], multipliers=pi)
-            else:
-                sol = self._createSolution(optTime, sol_inform, obj, xs[:nvar], multipliers=pi)
+            if parse_version(self.version) > parse_version("7.7.0") and parse_version(self.version) < parse_version("7.7.7"):
+                # SNOPT obj value is buggy and returned as 0, its thus overwritten with the solution objective value
+                obj = np.array([obj.value*obj.scale for obj in self.optProb.objectives.values()])
+
+            sol = self._createSolution(optTime, sol_inform, obj, xs[:nvar], multipliers=pi)
             restartDict = {
                 "cw": cw,
                 "iw": iw,

--- a/tests/testing_utils.py
+++ b/tests/testing_utils.py
@@ -119,16 +119,16 @@ class OptTest(unittest.TestCase):
         else:
             # assume we have a single solution
             self.sol_index = 0
+
         # now we assert against the closest solution
         # objective
         sol_objectives = np.array([obj.value for obj in sol.objectives.values()])
-        if self.optName == "SNOPT" and parse_version(self.optVersion) < parse_version("7.7.7"):
-            # don't check sol.fStar because it was broken for earlier versions of SNOPT
-            assert_allclose(sol_objectives, self.fStar[self.sol_index], atol=tol, rtol=tol)
-        else:
-            assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
-            # make sure fStar and sol.objectives values match
-            assert_allclose(sol.fStar, sol_objectives, rtol=1e-12)
+        assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
+        # make sure fStar and sol.objectives values match
+        # NOTE this is not true in general, but true for well-behaving optimizations
+        # which should be the case for all tests
+        assert_allclose(sol.fStar, sol_objectives, rtol=1e-12)
+
         # x
         assert_dict_allclose(sol.xStar, self.xStar[self.sol_index], atol=tol, rtol=tol, partial=partial_x)
         dv = sol.getDVs()

--- a/tests/testing_utils.py
+++ b/tests/testing_utils.py
@@ -6,6 +6,7 @@ import unittest
 from baseclasses.testing.assertions import assert_dict_allclose, assert_equal
 import numpy as np
 from numpy.testing import assert_allclose
+from pkg_resources import parse_version
 
 # First party modules
 from pyoptsparse import OPT, History
@@ -120,9 +121,14 @@ class OptTest(unittest.TestCase):
             self.sol_index = 0
         # now we assert against the closest solution
         # objective
-        assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
-        # make sure fStar and sol.objectives values match
-        assert_allclose(sol.fStar, [obj.value for obj in sol.objectives.values()], rtol=1e-12)
+        sol_objectives = np.array([obj.value for obj in sol.objectives.values()])
+        if self.optName == "SNOPT" and parse_version(self.optVersion) < parse_version("7.7.7"):
+            # don't check sol.fStar because it was broken for earlier versions of SNOPT
+            assert_allclose(sol_objectives, self.fStar[self.sol_index], atol=tol, rtol=tol)
+        else:
+            assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
+            # make sure fStar and sol.objectives values match
+            assert_allclose(sol.fStar, sol_objectives, rtol=1e-12)
         # x
         assert_dict_allclose(sol.xStar, self.xStar[self.sol_index], atol=tol, rtol=tol, partial=partial_x)
         dv = sol.getDVs()

--- a/tests/testing_utils.py
+++ b/tests/testing_utils.py
@@ -6,7 +6,6 @@ import unittest
 from baseclasses.testing.assertions import assert_dict_allclose, assert_equal
 import numpy as np
 from numpy.testing import assert_allclose
-from pkg_resources import parse_version
 
 # First party modules
 from pyoptsparse import OPT, History
@@ -122,11 +121,11 @@ class OptTest(unittest.TestCase):
 
         # now we assert against the closest solution
         # objective
-        sol_objectives = np.array([obj.value for obj in sol.objectives.values()])
         assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
         # make sure fStar and sol.objectives values match
         # NOTE this is not true in general, but true for well-behaving optimizations
         # which should be the case for all tests
+        sol_objectives = np.array([obj.value for obj in sol.objectives.values()])
         assert_allclose(sol.fStar, sol_objectives, rtol=1e-12)
 
         # x

